src/app/login/components/connectivityPane.tsx

```diff
@@ -29,22 +30,26 @@ export interface ConnectivityPaneDataProps {
 
 export interface ConnectivityState {
     connectionString: string;
+    connectionStringError: string;
     connectionStringList: string[];
-    error: string;
 }
 
 export default class ConnectivityPane extends React.Component<RouteComponentProps & ConnectivityPaneDataProps & ConnectivityPaneDispatchProps, ConnectivityState> {
     constructor(props: RouteComponentProps & ConnectivityPaneDataProps & ConnectivityPaneDispatchProps) {
         super(props);
+
+        const selectedConnectionString = this.props.connectionString ||
+            (this.props.connectionStringList && this.props.connectionStringList.length > 0 && this.props.connectionStringList[0]) || '';
```

Comment:
I'd like to avoid entangling the two in the store.  We'll actually need to phase out many of the direct connection string references to enable sso.  Also, there is no guarantee that connectionstring[0] will match the active resource set in the url.  So we could get into situations where the url is 'rickshub' with a connection string matching another hub.